File: FLLandBaseViewController(OC)/FLLandBaseViewController(OC)/FLLandBaseViewController.m
/*
 * author 孔凡列
 *
 * gitHub https://github.com/gitkong
 * cocoaChina http://code.cocoachina.com/user/
 * 简书 http://www.jianshu.com/users/fe5700cfb223/latest_articles
 * QQ 279761135
 * 喜欢就给个like 和 star 喔~
 */
#import "FLLandBaseViewController.h"

@interface FLLandBaseViewController ()

@end

@implementation FLLandBaseViewController

- (void)loadView{
    UIView *fl_view = [[UIView alloc] initWithFrame:CGRectMake(0, 0, [UIScreen mainScreen].bounds.size.height, [UIScreen mainScreen].bounds.size.width)];
    fl_view.backgroundColor = [UIColor whiteColor];
    self.view = fl_view;
}

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    //横屏.
    [self fl_screenStyle:UIInterfaceOrientationLandscapeRight];
}

- (void)viewWillDisappear:(BOOL)animated{
    [super viewWillDisappear:animated];
    //竖屏.
    [self fl_screenStyle:UIInterfaceOrientationPortrait];
}

#pragma mark -- private method
- (void)fl_screenStyle:(UIInterfaceOrientation)interfaceOrientation{
    NSNumber *value = [NSNumber numberWithInt:interfaceOrientation];
    [[UIDevice currentDevice] setValue:value forKey:@"orientation"];
}

@end

File: FLLandBaseViewController(Swift)/FLLandBaseViewController(Swift)/FLLandBaseViewController.swift
//
//  FLLandBaseViewController.swift
//  FLLandBaseViewController(Swift)
//
//  Created by clarence on 16/9/9.
//  Copyright © 2016年 clarence. All rights reserved.
//

import UIKit

class FLLandBaseViewController: UIViewController {
    
    override func loadView() {
        let fl_view = UIView.init(frame: CGRect.init(x: 0.0, y: 0.0, width: UIScreen.mainScreen().bounds.size.height, height: UIScreen.mainScreen().bounds.size.width))
        fl_view.backgroundColor = UIColor.whiteColor()
        self.view = fl_view
    }

    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        fl_screenStyle(UIInterfaceOrientation.LandscapeRight)
    }

    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        fl_screenStyle(UIInterfaceOrientation.Portrait)
    }
    
    func fl_screenStyle(style : UIInterfaceOrientation) {
        var variable : Int = 0
        switch style {
        case .Unknown:
            variable = 0
        case .Portrait:
            variable = 1
        case .PortraitUpsideDown:
            variable = 2
        case .LandscapeLeft:
            variable = 3
        default:
            variable = 4
        }
        let value : NSNumber = NSNumber.init(long: variable)
        UIDevice.currentDevice().setValue(value, forKey: "orientation")
    }
}
